src/artemis/fast_grid_scan_plan.py

```diff
@@ -75,36 +80,27 @@ def do_fgs():
         yield from bps.kickoff(fgs_motors)
         yield from bps.complete(fgs_motors, wait=True)
 
-    with ispyb as ispyb_ids, NexusWriter(
-        create_parameters_for_first_file(parameters)
-    ), NexusWriter(create_parameters_for_second_file(parameters)):
-        datacollection_ids = ispyb_ids[0]
-        datacollection_group_id = ispyb_ids[2]
-        for id in datacollection_ids:
-            run_start(id)
-        yield from do_fgs()
-
-    for id in datacollection_ids:
-        run_end(id)
-
-    xray_centre = wait_for_result(datacollection_group_id)
-    xray_centre_motor_position = (
-        parameters.grid_scan_params.grid_position_to_motor_position(xray_centre)
-    )
+    yield from do_fgs()
+    yield from bps.abs_set(fgs_motors.z_steps, 0, wait=False)
 
-    yield from bps.mv(
-        sample_motors.x,
-        xray_centre_motor_position.x,
-        sample_motors.y,
-        xray_centre_motor_position.y,
-        sample_motors.z,
-        xray_centre_motor_position.z,
-    )
 
-    yield from bps.abs_set(fgs_motors.z_steps, 0, wait=False)
+def run_gridscan_and_move(
+    fgs_composite: FGSComposite,
+    eiger: EigerDetector,
+    parameters: FullParameters,
+    communicator: FGSCommunicator,
+):
+    @subs_decorator(communicator)
+    def decorated():
+        yield from run_gridscan(fgs_composite, eiger, parameters)
+        yield from move_xyz(
+            fgs_composite.sample_motors, communicator.xray_centre_motor_position
+        )
 
+    yield from decorated()
```

Comment:
Should: As discussed I think something a little more like:
```python
    yield from bps.subscribe("all", communicator)
    yield from run_gridscan(fgs_composite, eiger, parameters)
    yield from bps.unsubscribe(communicator)
    
    communicator.wait_for_results()
    yield from move_xyz(
        fgs_composite.sample_motors, communicator.xray_centre_motor_position
    )
```
Gets round having the communicator have to check what plan is running and is more explicit in when results are being waited on.